src/selectRace/races.ts

```diff
@@ -1,51 +1,45 @@
-import getAllRuns, { getAllLapsPerRun } from "../api/requests";
+import {
+  allFilesSubject$,
+  runSummarySubject$,
+  signalNewFilenameRequest$,
+  signalNewLapsPerRunRequest$,
+} from "../api/requests";
 import { createRaceCard, hideSpinner, showSpinner } from "../dom/ui-manip";
 import "../index.scss";
-import { IKartLapsPerRun } from "../models/go-kart-types";
 
-let filename;
+function getData() {
+  showSpinner();
+  signalNewFilenameRequest$.next(null);
+}
```

Comment:
See comment above with regards to the null and the observable. If it is continiously being set to null, is it not then redundant?

Reply:
Not being set to null just receiving a null parameter.